### ISSUE TITLE
Gene usage plot

### ISSUE THREAD

**Issue by crushseven-7** [immunomind/immunarch#95]
Hi there,
I am working on 10X scTCR data lately. Here I got a problem, after I loaded my 10X data into immunarch, it would store paired info (alpha and beta chain) like the picture blow  
![paired info](https://s1.ax1x.com/2020/08/05/asuwZV.png "paired info")  
Consequently, when I caculate the V/J gene usage, the x axis would show all V/J combination rather than each single V/J gene usage which in my opinion using hitmap can get a more intuitive display.  
![usage result](https://s1.ax1x.com/2020/08/05/asKEwV.png "usage result")  
So is there any tricky way to show one gene in one column or else to handle the input before plot.

**Comment by vadimnazarov** [comment]
Hi @crushseven-7 

Thank you opening a ticket!

In case of the heatmap, what would be the output for the input groups of repertoires? Heatmap would work greatly on a single repertoires, but what if you get a group of them?

Would you like to count occurrences separately for V and J genes? What would you like to see the results of the plot – two plots one for V and for J? Or would you like to choose what gene to count?

**Comment by crushseven-7** [comment]
Thank you for your response. As you said, heatmep may just suitable for a single sample. Maybe plot a heatmap for each repertories?
As for geneusage, as shown in the below picture. In the `geneusage` function, I fill up with 'hs.trav' in the `.gene` parameter. Consequently I would expect each row would be one kind of trav gene in the output dataframe. In this way, the plot will be better looking.
![geneUsage result](https://ftp.bmp.ovh/imgs/2020/08/03882d858a689cb1.png "geneUsage result")

**Comment by kalinabelcheva** [comment]
I agree with crushseven-7.  It would be really useful to plot the single V/J gene usage instead of the combinations. My analysis also included "None" as part of the combinations (e.g. bottom right on the image). That is worth fixing.  

Sometimes the sequencing data lacks all the combinations of light+heavy (BCR) or alpha+beta(TCR) chains which precludes the interpretation of such combination plots. 

<img width="1402" alt="Screen Shot 2020-09-29 at 9 07 32 AM" src="https://user-images.githubusercontent.com/72039044/94562421-5229f080-0233-11eb-8ba8-44e28722afea.png">

Immunarch looks great and I would be happy to use it in the future once this is fixed.

**Comment by linqy-immune** [comment]
> I agree with crushseven-7. It would be really useful to plot the single V/J gene usage instead of the combinations. My analysis also included "None" as part of the combinations (e.g. bottom right on the image). That is worth fixing.
> 
> Sometimes the sequencing data lacks all the combinations of light+heavy (BCR) or alpha+beta(TCR) chains which precludes the interpretation of such combination plots.
> 
> <img alt="Screen Shot 2020-09-29 at 9 07 32 AM" width="1402" src="https://user-images.githubusercontent.com/72039044/94562421-5229f080-0233-11eb-8ba8-44e28722afea.png">
> 
> Immunarch looks great and I would be happy to use it in the future once this is fixed.

Hello! I encountered this problem when plotting the gene usage of several repertoires. How did you revise the code to plot the single V/J gene usage instead of the combinations? Could you please give me some suggestions?

**Comment by linqy-immune** [comment]
> Thank you for your response. As you said, heatmep may just suitable for a single sample. Maybe plot a heatmap for each repertories? As for geneusage, as shown in the below picture. In the `geneusage` function, I fill up with 'hs.trav' in the `.gene` parameter. Consequently I would expect each row would be one kind of trav gene in the output dataframe. In this way, the plot will be better looking. ![geneUsage result](https://camo.githubusercontent.com/783dc940c1536b2c1dafb1d41130002c7aa6db37cdec3bc55ecfb7acf0a307b1/68747470733a2f2f6674702e626d702e6f76682f696d67732f323032302f30382f303338383264383538613638396362312e706e67)
@crushseven-7 
Hello! I encountered this problem when plotting the gene usage of several repertoires. How did you revise the code to plot the single V/J gene usage instead of the combinations? Could you please give me some suggestions? Thanks and best regards!
